src/utilities/createRouterRoute.ts

```diff
@@ -2,49 +2,35 @@ import { DeepReadonly, reactive, readonly } from 'vue'
 import { Resolved, Route } from '@/types'
 import { RouterPushError, RouterRejectionError, RouterReplaceError } from '@/types/errors'
 import { RouterPushImplementation } from '@/utilities/createRouterPush'
-import { CreateRouterReject, RouterReject } from '@/utilities/createRouterReject'
+import { RouterReject, notFoundRouteResolved } from '@/utilities/createRouterReject'
 import { RouterReplaceImplementation } from '@/utilities/createRouterReplace'
 import { RouterResolveImplementation } from '@/utilities/createRouterResolve'
-import { getInitialUrl } from '@/utilities/getInitialUrl'
-import { routeMatch } from '@/utilities/routeMatch'
 import { RouterNavigation } from '@/utilities/routerNavigation'
 import { getRouteMiddleware } from '@/utilities/routes'
 
-type RouterRouteUpdate = (url: string) => Promise<void>
+type RouterRouteUpdate = (matched: Resolved<Route>) => void
+type RouterRouteMiddleware = (matched: Resolved<Route>) => Promise<void>
 
 type RouterRoute = {
   route: DeepReadonly<Resolved<Route>>,
   updateRoute: RouterRouteUpdate,
-  initialized: Promise<void>,
+  executeMiddleware: RouterRouteMiddleware,
 }
 
-type RouterGetContext = {
-  resolved: Resolved<Route>[],
-  resolve: RouterResolveImplementation,
-  navigation: RouterNavigation,
-  routerReject: CreateRouterReject,
-  initialUrl: string | undefined,
-}
-
-export function createRouterRoute({ resolved, resolve, navigation, routerReject, initialUrl }: RouterGetContext): RouterRoute {
-  const { reject, getRejectionRoute, clearRejection } = routerReject
-  const route = reactive<Resolved<Route>>(getRejectionRoute('NotFound'))
+export function createRouterRoute(resolve: RouterResolveImplementation, navigation: RouterNavigation): RouterRoute {
+  const route = reactive<Resolved<Route>>(notFoundRouteResolved)
 
   const setRoute = (newRoute: Resolved<Route>): void => {
     Object.assign(route, newRoute)
   }
 
-  const updateRoute: RouterRouteUpdate = async (url) => {
-    const matched = routeMatch(resolved, url)
-
-    if (!matched) {
-      reject('NotFound')
-      setRoute(getRejectionRoute('NotFound'))
-      return
-    }
+  const updateRoute: RouterRouteUpdate = (matched: Resolved<Route>) => {
+    setRoute(matched)
+  }
 
+  const executeMiddleware: RouterRouteMiddleware = async (matched) => {
```

Comment:
We should just pull this out and pass the current route into it as another arg. Then this can be its own utility.

Reply:
agreed. I think you resolved this in your PR?

Reply:
Yep I did